### PR TITLE
OCPBUGS-17558 & 17559: Adding NFS support to Azure File CSI

### PR DESCRIPTION
**Pull request by lpettyjo**
Version(s): 4.12 & 4.13

Issue: https://issues.redhat.com/browse/OCPBUGS-17558 and https://issues.redhat.com/browse/OCPBUGS-17559

Link to docs preview: https://63854--docspreview.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-azure-file#persistent-storage-csi-azure-file-nfs_persistent-storage-csi-azure-file

QE review:
- [x] QE has approved this change.

Additional information:

PR to add Known Issue to RN 4.12: https://github.com/openshift/openshift-docs/pull/65713

**PTAL**: @bertinatto, @duanwei33, @gcharot, @jsafrane, @sferich888, @xltian, @mwerner2113

### CODE REVIEW

**Review by kelbrown20 — COMMENTED** [review]
Just some suggestions and nits. Otherwise, this looks good to me!

**Comment by jsafrane** [comment]
LGTM

**Comment by mwerner2113** [comment]
lgtm

**Comment by kalexand-rh** [comment]
@lpettyjo, there's a [fresh bug](https://issues.redhat.com/browse/OCPBUGS-18446) saying that "Azure File" should be "Azure Files." Will you please confirm?

File: storage/container_storage_interface/persistent-storage-csi-azure-file.adoc [hunk 1]
```diff
@@ -21,12 +21,12 @@ Azure File CSI Driver Operator does not support:
 
 * Virtual hard disks (VHD)
 
-* Network File System (NFS): {product-title} does not deploy a NFS-backed storage class.
-
-* Running on nodes with FIPS mode enabled.
+* Running on nodes with Federal Information Processing Standard (FIPS) mode enabled for Server Message Block (SMB) file share. However, Network File System (NFS) does support FIPS mode.
```
Comment (sferich888) [inline storage/container_storage_interface/persistent-storage-csi-azure-file.adoc:24]: Do we need to say anything about SMB (thats a different FS type) ?

**Review by bertinatto — COMMENTED** [review]
LGTM

**Comment by xltian** [comment]
lgtm

File: storage/container_storage_interface/persistent-storage-csi-azure-file.adoc (hunk 1 above)
Comment (lpettyjo) [reply storage/container_storage_interface/persistent-storage-csi-azure-file.adoc:24]: I've added an emphasis on the "not" to make this clearer.

**Comment by gcharot** [comment]
This microsoft driver is called Azure File CSI driver so calling it "Azure Files" is incorrect.
https://github.com/kubernetes-sigs/azurefile-csi-driver

The same applies to our operator.

Now for occurences where we mention the Microsoft Azure service itself and not the driver or operator. I am open to change it.
For example:
`OpenShift Container Platform is capable of provisioning persistent volumes (PVs) by using the Container Storage Interface (CSI) driver for Microsoft Azure File Storage.`
Here we're talking about the Azure Files service not the driver.

`Azure File CSI Driver Operator does not support:`
Here we're talking about the operator so it must be singular.

That being said, mixing singular and plural can bring more confusion than anything.

File: modules/persistent-storage-csi-azure-file-nfs.adoc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+[id="persistent-storage-csi-azure-file-nfs_{context}"]
+= NFS support
+
+{product-title} supports the Azure File Container Storage Interface (CSI) Driver Operator with Network File System (NFS) with the following restrictions:
+
+* Creating pods with Azure File NFS volumes that are scheduled to the control plane node causes the mount to be denied.
++
+To work around this issue:
+
+** If the control plane nodes are schedulable, and the pod with Azure File NFS volumes could be scheduled to worker nodes, use nodeSelector or Affinity to schedule the pod in worker nodes.
```
Comment (gcharot) [inline modules/persistent-storage-csi-azure-file-nfs.adoc:16]: Azure File NFS volumes could be scheduled to control plane nodes?

File: modules/persistent-storage-csi-azure-file-nfs.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+** If there are no worker nodes, or you need to schedule the pod with Azure File NFS volume to the control plane nodes, update the storage account (used to create the Azure File NFS volume) in the Azure portal, select networking, and then ensure that the control plane node's VNET and subnet have access.
```
[the remark this answers is not in the document]
Comment (lpettyjo) [reply modules/persistent-storage-csi-azure-file-nfs.adoc:18]: The second option has been removed.

**Comment by gcharot** [comment]
LGTM

**Pull request merged**